### PR TITLE
Make SymfonyContainer available from 1 location

### PR DESCRIPTION
**Pull request by aleeks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Make SymfonyContainer available from 1 location
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | more easily to use SymfonyContainer from legacy...

ping @Quetzacoalt91 @maximebiloe @eternoendless 

to update **after** this is merged: 
- #7919 
- #8215

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
If merged, #7919 will need an update
and https://github.com/PrestaShop/PrestaShop/pull/8215

File: classes/SymfonyContainer.php
Comment (eternoendless) [inline classes/SymfonyContainer.php]: Please add a class header explaining the purpose of this class

File: classes/helper/HelperList.php
Comment (Quetzacoalt91) [inline classes/helper/HelperList.php]: You should not have to check, because we are in the admin environment, so `$kernel` must be defined

File: classes/SymfonyContainer.php
Comment (eternoendless) [inline classes/SymfonyContainer.php]: There's one * missing in `/*` (to enable PHPDoc parsing)

Comment (eternoendless) [reply classes/SymfonyContainer.php]: You can also use `self` instead of `SymfonyContainer`

File: classes/SymfonyContainer.php
Comment (eternoendless) [inline classes/SymfonyContainer.php]: I don't think that `Context` is the right return type

File: controllers/admin/AdminThemesController.php
Comment (Quetzacoalt91) [inline controllers/admin/AdminThemesController.php]:
```php
$exporter = SymfonyContainer::getSfContainer()->get('prestashop.core.addon.theme.exporter');
```

File: classes/SymfonyContainer.php
Comment (eternoendless) [inline classes/SymfonyContainer.php]: You should check against `KernelInterface` instead (to ensure the presence of `->getContainer()`)

File: classes/SymfonyContainer.php
Comment (eternoendless) [inline classes/SymfonyContainer.php]: Since it's a singleton instance, I'd rather go for "getInstance"

File: classes/SymfonyContainer.php
Comment (eternoendless) [inline classes/SymfonyContainer.php]: This should throw an Exception if not in Symfony context

File: classes/SymfonyContainer.php
Comment (Quetzacoalt91) [inline classes/SymfonyContainer.php]: This time, why don't you write the namespace at the top of the file? :wink:

**Review by eternoendless — CHANGES_REQUESTED** [review]

File: classes/SymfonyContainer.php
Comment (Quetzacoalt91) [inline classes/SymfonyContainer.php]: Be careful here. I saw that even if sometimes you do not have the current request, the Symfony kernel can be used anyway.

File: classes/SymfonyContainer.php
Comment (Quetzacoalt91) [inline classes/SymfonyContainer.php]: Can be removed

File: classes/SymfonyContainer.php
Comment (Quetzacoalt91) [inline classes/SymfonyContainer.php]: Because you will return it everytime, you can affect a default value like `null`.

File: classes/helper/HelperList.php
Comment (Quetzacoalt91) [inline classes/helper/HelperList.php]: You should not have to check, because we are in the admin environment, so `$kernel` must be defined

**Review by Quetzacoalt91 — CHANGES_REQUESTED** [review]
Here are some comments after a quick read.

File: classes/SymfonyContainer.php
Comment (aleeks) [reply classes/SymfonyContainer.php]: hihi, it's issue from Context, maybe the /** var is missing to :smile:

File: classes/SymfonyContainer.php
Comment (eternoendless) [reply classes/SymfonyContainer.php]: Not if private

Comment (eternoendless) [reply classes/SymfonyContainer.php]: In any case, `{` `}` should go in a new line per PSR-2.

**Comment by aleeks** [comment]
@eternoendless @Quetzacoalt91 can you check again please ?!

**Review by Quetzacoalt91 — APPROVED** [review]
On my side: OK
Waiting for another review before merge.

**Review by eternoendless — APPROVED** [review]
Looks good to me!

**Comment by aleeks** [comment]
It's on develop, if it creates error, we have time to find them :smile_cat:

**Comment by eternoendless** [comment]
Thanks @aleeks !

**Pull request merged**
